buildpacks/jvm-function-invoker/src/layers/opt.rs

```diff
@@ -1,55 +1,55 @@
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use libcnb::layer::{Layer, LayerResult, LayerResultBuilder};
+
 use std::fs;
 use std::os::unix::fs::PermissionsExt;
-use std::path::{Path, PathBuf};
-
-use libcnb::data::layer_content_metadata::LayerContentMetadata;
-use libcnb::layer_lifecycle::LayerLifecycle;
-use libcnb::{BuildContext, GenericMetadata, GenericPlatform};
+use std::path::Path;
 
 use crate::error::JvmFunctionInvokerBuildpackError;
-use crate::JvmFunctionInvokerBuildpackMetadata;
-
-pub struct OptLayerLifecycle {}
-
-#[cfg(target_family = "unix")]
-impl
-    LayerLifecycle<
-        GenericPlatform,
-        JvmFunctionInvokerBuildpackMetadata,
-        GenericMetadata,
-        PathBuf,
-        JvmFunctionInvokerBuildpackError,
-    > for OptLayerLifecycle
-{
-    fn create(
-        &self,
-        path: &Path,
-        context: &BuildContext<GenericPlatform, JvmFunctionInvokerBuildpackMetadata>,
-    ) -> Result<LayerContentMetadata<GenericMetadata>, JvmFunctionInvokerBuildpackError> {
-        let source = context.buildpack_dir.join("opt").join("run.sh");
-        let destination = path.join("run.sh");
+use crate::JvmFunctionInvokerBuildpack;
 
-        fs::copy(&source, &destination).map_err(OptLayerError::CouldNotCopyRunSh)?;
+pub struct OptLayer;
 
-        fs::set_permissions(&destination, fs::Permissions::from_mode(0o755))
-            .map_err(OptLayerError::CouldNotSetExecutableBitForRunSh)?;
+impl Layer for OptLayer {
+    type Buildpack = JvmFunctionInvokerBuildpack;
+    type Metadata = GenericMetadata;
 
-        Ok(LayerContentMetadata::default().launch(true))
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            launch: true,
+            build: false,
+            cache: false,
+        }
     }
 
-    fn layer_lifecycle_data(
+    fn create(
         &self,
-        path: &Path,
-        _layer_content_metadata: LayerContentMetadata<GenericMetadata>,
-    ) -> Result<PathBuf, JvmFunctionInvokerBuildpackError> {
-        Ok(path.join("run.sh"))
+        _context: &BuildContext<Self::Buildpack>,
+        layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, JvmFunctionInvokerBuildpackError> {
+        let layer_bin_dir = layer_path.join("bin");
+        let destination = layer_bin_dir.join(JVM_RUNTIME_SCRIPT_NAME);
+
+        fs::create_dir_all(&layer_bin_dir).map_err(OptLayerError::CouldNotWriteRuntimeScript)?;
+
+        fs::write(&destination, include_bytes!("../../opt/jvm-runtime.sh"))
```

Comment:
This changed from copying a file from the buildpack directory since `libcnb-cargo` cannot package additional files currently. IMO, using `include_bytes!` is better anyway since it removes some potential errors at runtime.